pkg/client/client.go

```diff
@@ -1215,37 +1209,31 @@ func (c *KpmClient) downloadDeps(deps pkg.Dependencies, lockDeps pkg.Dependencie
 			if os.IsNotExist(err) {
 				continue
 			}
-			return nil, nil, err
-		}
-
-		// Download the dependencies.
-		nested, nestedDepGraph, err := c.downloadDeps(deppkg.ModFile.Dependencies, lockDeps)
-		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 
 		source := fmt.Sprintf("%s@%s", d.Name, d.Version)
 		err = depGraph.AddVertex(source)
 		if err != nil && err != graph.ErrVertexAlreadyExists {
-			return nil, nil, err
+			return nil, err
 		}
 
-		sourcesOfNestedDepGraph, err := pkgGraph.FindSources(nestedDepGraph)
+		err = depGraph.AddEdge(parent, source)
 		if err != nil {
-			return nil, nil, err
+			if err == graph.ErrEdgeCreatesCycle {
+				return nil, reporter.NewErrorEvent(
+					reporter.CircularDependencyExist,
+					nil,
+					"adding dependencies results in a cycle",
```

Comment:
Suggest adding which packages have circular references to the error message to help developers improve dependencies.

Reply:
done!